resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -116,7 +116,10 @@ private[spark] class KubernetesDriverConf(
   }
 
   override def annotations: Map[String, String] = {
-    KubernetesUtils.parsePrefixedKeyValuePairs(sparkConf, KUBERNETES_DRIVER_ANNOTATION_PREFIX)
+    KubernetesUtils.parsePrefixedKeyValuePairs(sparkConf, KUBERNETES_DRIVER_ANNOTATION_PREFIX).map {
+      case(k, v) =>
+        (k, Utils.substituteAppNExecIds(v, appId, ""))
+    }
```

Comment:
If you don't mind, shall we use the consistent short style like `KubernetesExecutorConf`, @zwangsheng ?